app/src/ui/toolbar/push-pull-button.tsx

```diff
@@ -273,51 +153,162 @@ const forcePushIcon: OcticonSymbol.OcticonSymbolType = {
   fr: 'evenodd',
 }
 
-function forcePushButton(
-  remoteName: string,
-  aheadBehind: IAheadBehind,
-  numTagsToPush: number,
-  lastFetched: Date | null,
-  onClick: () => void
-) {
-  return (
-    <ToolbarButton
-      {...defaultProps}
-      title={`Force push ${remoteName}`}
-      description={renderLastFetched(lastFetched)}
-      icon={forcePushIcon}
-      onClick={onClick}
-    >
-      {renderAheadBehind(aheadBehind, numTagsToPush)}
-    </ToolbarButton>
-  )
-}
-
 /**
  * A button which pushes, pulls, or updates depending on the state of the
  * repository.
  */
-export class PushPullButton extends React.Component<IPushPullButtonProps, {}> {
+export class PushPullButton extends React.Component<IPushPullButtonProps> {
+  /** The common props for all button states */
+  private defaultButtonProps() {
+    return {
+      className: 'push-pull-button',
+      style: ToolbarButtonStyle.Subtitle,
+    }
+  }
+
+  /** The common props for all dropdown states */
+  private defaultDropdownProps(): Omit<
+    IToolbarDropdownProps,
+    'dropdownContentRenderer'
+  > {
+    return {
+      buttonClassName: 'push-pull-button',
+      style: ToolbarButtonStyle.Subtitle,
+      dropdownStyle: ToolbarDropdownStyle.MultiOption,
+      dropdownState: this.props.isDropdownOpen ? 'open' : 'closed',
+      onDropdownStateChanged: this.props.onDropdownStateChanged,
+    }
+  }
+
+  private closeDropdown() {
+    this.props.dispatcher.closeFoldout(FoldoutType.PushPull)
+  }
+
   private push = () => {
+    this.closeDropdown()
     this.props.dispatcher.push(this.props.repository)
   }
 
   private forcePushWithLease = () => {
+    this.closeDropdown()
     this.props.dispatcher.confirmOrForcePush(this.props.repository)
   }
 
   private pull = () => {
+    this.closeDropdown()
     this.props.dispatcher.pull(this.props.repository)
   }
 
   private fetch = () => {
+    this.closeDropdown()
     this.props.dispatcher.fetch(
       this.props.repository,
       FetchType.UserInitiatedTask
     )
   }
 
+  private getDropdownContentRenderer(
+    itemTypes: ReadonlyArray<DropdownItemType>
+  ) {
+    return () => {
+      return (
+        <>
+          {/*
+          This <div> is not interactive, but its children are, so we need to
+          disable the jsx-a11y/no-static-element-interactions rule.
+          */}
+          {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions */}
+          <div
```

Comment:
I had a similar problem with the items of the `DropdownSelectButton` needing to be up and down arrow navigable. Did a bit of looking around at other implementations on the web to make it accessible. Came to the conclusion that using Buttons instead of divs is more accessible. You get the up/down navigation for free and no disabling linters.  The negative about it is that these do not feel like buttons -> They feel like list items and will need styled to look like list items in this approach. Thus, it feels like it breaks the html convention of using the elements for what they present as. But, I think it is arguable that these are actually buttons because you click on them and expect an action just like a button. Thus we should be using an interact-able html element -> a button. 

However, I see you are doing some stuff with focusing and I am not sure this will solve all that, but I think that the onKeyDown will no longer complain to the linter for a button element if the focus logic is still needed.

Reply:
Hmm.. I think I may have misunderstood the code... cause the individual items are buttons.. disregard above. Still think we should try to find a way not to introduce more jsx-ally linters, but till I understand why it's needed I am not sure how to remove it. :D

Reply:
Exactly, this is a wrapper with buttons inside of them. I can try to bubble those buttons up and see if it doesn't break the style 😳

Reply:
Alright.. so I did have this problem before.. :D I just was looking at the wrong part. Check out `avoiding-disabling-the-linter` to see how I went about it another way to not have to introduce a linter disabling.. (not sure if avoiding the linter is the best reason to be doing this.. but it also separates the concern out a little).

Reply:
I love it, thank you!! ❤️ 

I don't know if the `window.addEventListener` is better or worse than `onKeyDown`, but definitely helps to avoid the lint issue 😄